src/xpi.js

```diff
@@ -134,28 +134,27 @@ export default class Xpi {
   }
 
   getFilesByExt(...extensions) {
-    return new Promise((resolve, reject) => {
 
-      for (let ext of extensions) {
-        if (ext.indexOf('.') !== 0) {
-          throw new Error("File extension must start with '.'");
-        }
+    for (let ext of extensions) {
+      if (ext.indexOf('.') !== 0) {
+        // Use Promise.reject as outside a then a throw
+        // wouldn't be caught.
+        return Promise.reject(new Error("File extension must start with '.'"));
```

Comment:
Reading the comment again, yeah it doesn't quite stand out.

```js
// We use Promise.reject as we're outside a `then` block.
// Using `throw` directly wouldn't be caught.
```
Might work better? I try to use markdown in comments to help highlight references to code.